### ISSUE TITLE
Redshift fails on startup

### ISSUE THREAD

**Issue by vra5107** [jonls/redshift#333]
I have been using redshift for quite a while now. Not sure what happened, but after I restarted the computer, it stopped working. 

Here is the console output when I run `redshift`.
```
venkat@venkat-U46E:~ > /usr/bin/redshift
Trying location provider `geoclue'...
Started Geoclue provider `Geoclue Master'.
Using provider `geoclue'.
Could not get location: xmlNanoHTTPMethod did not get a response from http://api.hostip.info/
.
Unable to get location from provider.
```

Here is the console output when I run `redshift-gtk`.
```

venkat@venkat-U46E:~ > redshift-gtk

** (redshift-gtk:24082): WARNING **: Error retrieving accessibility bus address: org.freedesktop.DBus.Error.ServiceUnknown: The name org.a11y.Bus was not provided by any .service files
```

I am not sure where redshift logs are stored so I am unable to attach them here.

**Comment by Peterdes** [comment]
The same here, Mint 17.3

**Comment by jetojedno** [comment]
Can I make a suggestion: the ability to add a fallback location (manual) to be used if geolocation from IP address or phone fails.

**Comment by vchryssos** [comment]
I use redshift 1.9.1 with linux Mint 17.3 KDE. I have the same problem with @vra5107. 
To overcome this I've added the following command as a startup application:
`redshift -l xx:yy`
where 
xx=lat, 
yy=lon.

This works for me.

**Comment by rugk** [comment]
Even with `-l xx:yy` it does not work for me.

The bad thing is when I run everything with `-v` it correctly tells me the location I set, but it still tries to request location updates. Sorry, but that's impossible on a PC... :neutral_face:

**Comment by rugk** [comment]
Even `-l manual -l xx:yy` fails.

**Comment by charlieb000** [comment]
Just installed trisquel and then installed redshift to try. fails for same reason. I am guessing it needs my location for latitude (in that case, cant i just enter it or surmise latitude from the time the sun actually sets? instead of just failing. btw, who else knows my location...?)

**Comment by rugk** [comment]
> (in that case, cant i just enter it or let it surmise the latitude from the time I say the sun actually sets at the particular day of the year?).

That's exactly what I suggest too. It is certainly possible, but RedShift has not implemented this.

**Comment by charlieb000** [comment]
On windows there is f.lux and I had issues with that too so i just use yellow glasses (plus i do not need to adjust the lights in the house so it looks normal outside - it is just that I look less normal!). The primary issue was with video overlay (and also the program seemed to be interested in my IE files...). I only wanted to see how a similar program performed under linux.

**Comment by rugk** [comment]
Yes, I know f.lux is also available (for Linux) too, but it is not open-source. :expressionless:

**Comment by rugk** [comment]
Any news about this issue, @jonls?
Redshift seems to always get some coordinates (although they're wrong) - but this does not matter as I can correct them in the manual mode. It just always shows me "No methods to try":

```console
$ redshift -vvv
Versuche Location Provider `geoclue2'...
Dienst »geoclue2« wird benutzt.
Position: 51.30° N, 12.33° O
Farbtemperatur: 5500K tagsüber, 3500K nachts
Solar elevations: day above 3.0, night below -6.0
Helligkeit: 1.00:1.00
Gamma (Tagsüber): 1.000, 1.000, 1.000
Gamma (Nachts): 1.000, 1.000, 1.000
Keine weiteren Methoden verfügbar.
$ redshift -l 51:32 -vvv
Position: 51.00° N, 32.00° O
Farbtemperatur: 5500K tagsüber, 3500K nachts
Solar elevations: day above 3.0, night below -6.0
Helligkeit: 1.00:1.00
Gamma (Tagsüber): 1.000, 1.000, 1.000
Gamma (Nachts): 1.000, 1.000, 1.000
Keine weiteren Methoden verfügbar.
```

**Comment by jonls** [comment]
@rugk Did you compile Redshift manually? Did you enable any adjustment methods during `./configure`?

**Comment by rugk** [comment]
No I am just using the version from Debian Jessie: https://packages.debian.org/jessie/redshift

**Comment by jonls** [comment]
From the output that you provided it looks like Redshift was not compiled with any adjustment methods. Have you tried reporting this problem to Debian?

**Comment by rugk** [comment]
What exactly does it mean "no adjustment method"? Can you point me to some doc or so?

**Comment by jonls** [comment]
Redshift has several ways of adjusting the gamma ramps: Vidmode, Randr, Quartz (on macOS), wingdi (on Windows), etc. When compiling, you have to specify which adjustment method you want to include in the the application. If you didn't include any, Redshift won't be able to do anything, it will just fail with "no adjustment methods left to try".

**Comment by rugk** [comment]
BTW in case you care: The error message shown to me is "Keine weiteren Methoden verfügbar.", which is German for "No more  methods available.". I don't know whether this is a translation issue, but the error message you quoted ("no adjustment methods left to try") would be much more appropriate and also much more detailed.

**Comment by rugk** [comment]
I have reported this issue to Debian.
